build_tools/cmake/iree_bytecode_module.cmake

```diff
@@ -98,21 +104,17 @@ function(iree_bytecode_module)
 
   # Check LLVM static library setting. If the static libary output path is set,
   # retrieve the object path and the corresponding header file path.
-  foreach(_FLAG ${_RULE_FLAGS})
-    string(REGEX MATCH
-      "-iree-llvm-static-library-output-path=(.+)" _MATCH_STRING "${_FLAG}"
-    )
-    if(_MATCH_STRING)
-      set(_STATIC_LIB TRUE)
-      # Get the static object path from the regex match result.
-      set(_STATIC_OBJ_PATH ${CMAKE_MATCH_1})
-      string(REPLACE ".o" ".h" _STATIC_HDR_PATH "${_STATIC_OBJ_PATH}")
-    endif()
-  endforeach(_FLAG)
+  if(_RULE_STATIC_LIB_PATH)
+    list(APPEND _ARGS "--iree-hal-target-backends=llvm-cpu")
```

Comment:
Huh I missed that we weren't already passing the backend as an explicit cmake arg. That makes this way of doing things a bit more weird (because now if I switch from non-static to static lib I drop the target arg). I still think it's better though

Reply:
I can leave the target backend flag outside of the `iree_*_module` macro, although in L53 we still want to check the setting condition as
```
if(_RULE_STATIC_LIB_PATH AND NOT IREE_TARGET_BACKEND_LLVM_CPU)
    message(SEND_ERROR "Static library only supports llvm-cpu backend")
  endif()
```

Reply:
Oh I wasn't saying change it, necessarily. What you have now and what you had before that last commit are both fine :-) I think the ideal given we're doing things based on backend would be to have the backend as an explicit arg also, but I'm not going to punish you with extra work just because you touched this rule :-P